src/main/scala/net/snowflake/spark/snowflake/io/CloudStorageOperations.scala

```diff
@@ -352,7 +358,10 @@ object CloudStorageOperations {
             param.useExponentialBackoff,
             param.expectedPartitionCount,
             pref = prefix,
-            connection = conn
+            connection = conn,
+            // For S3 external stage, it doesn't use region name in URL
```

Comment:
why?

Reply:
For external stage, we can't determine the region name automatically, it needs user configure it.
But external stage is legacy, it is not necessary to introduce such a `feature`.